avro2tf/src/main/scala/com/linkedin/avro2tf/jobs/PrepRankingData.scala

```diff
@@ -187,20 +209,22 @@ object PrepRankingData {
   }
 
   /**
-   * udf to flatten the 2d sparse tensor to 1d.
+   * udf to have 2d Sparse Vector. The input is expected to be [[SparseVector]] with indices and values
    */
-  private val flattenSparseVectorArray =
+  private def multiDimSparseVectorArray(outputSchema: StructType) =
```

Comment:
This method is hard-coded for 2 dimensions. This means that we're always passing the same schema as an argument. For consumers of this method, it would be simpler and cleaner if we didn't take `outputSchema` as an argument. We could call `generateSchemaForMultiDimSparseFeature(2)` inside this method body instead.

Reply:
good suggestion. Just add some background, Avro2TF can only handle 1D sparse input, after groupby will always be 2D.